h2o-algos/src/main/java/hex/glm/GLM.java

```diff
@@ -847,11 +902,81 @@ private GLMModel runGLMModel(boolean standardize, Family family, Link link, Key<
     GLMModel model = new GLM(tempParams).trainModel().get();
     return model;
   }
+  
+  // copy from scoring_history back to _sc or _lsc
+  private void reinstallSC() {
+      TwoDimTable scoringHistory = _model._output._scoring_history;
+      String[] colHeaders2Restore = _parms._lambda_search ? 
+              new String[]{"iteration", "timestamp", "lambda", "predictors", "deviance_train", 
+                      "deviance_test"}
+              : new String[]{"iteration", "timestamp", "negative_log_likelihood", "objective", "sum(etai-eta0)^2", 
+              "convergence"};
+      int num2Copy = _parms._HGLM || _parms._lambda_search ? colHeaders2Restore.length : colHeaders2Restore.length-2;
+      int[] colHeadersIndex = grabHeaderIndex(scoringHistory, num2Copy, colHeaders2Restore);
+      if (_parms._lambda_search)
+        restoreLSC(scoringHistory, colHeadersIndex);
+      else
+        restoreSC(scoringHistory, colHeadersIndex);
+  }
+  
+  private void restoreLSC(TwoDimTable sHist, int[] colIndices) {
+    DateTimeFormatter fmt = DateTimeFormat.forPattern("yyyy-MM-dd HH:mm:ss");
+    int numRows = sHist.getRowDim();
+    for (int rowInd = 0; rowInd < numRows; rowInd++) {
+      _lsc._scoringTimes.add(fmt.parseMillis((String) sHist.get(rowInd, colIndices[1])));
+      _lsc._lambdaIters.add((int) sHist.get(rowInd, colIndices[0]));
+      _lsc._lambdas.add(Double.valueOf((String) sHist.get(rowInd, colIndices[2])));
+      _lsc._lambdaPredictors.add((int) sHist.get(rowInd, colIndices[3]));
+      _lsc._lambdaDevTrain.add((double) sHist.get(rowInd, colIndices[4]));
+      _lsc._lambdaDevTest.add((double) sHist.get(rowInd, colIndices[5]));
+    }
+  }
+
+  private void restoreSC(TwoDimTable sHist, int[] colIndices) {
+    DateTimeFormatter fmt = DateTimeFormat.forPattern("yyyy-MM-dd HH:mm:ss");
+    int numRows = sHist.getRowDim();
+    for (int rowInd = 0; rowInd < numRows; rowInd++) {  // if lambda_search is enabled, _sc is not updated
+      _sc._scoringIters.add((Integer) sHist.get(rowInd, colIndices[0]));
+//      _scoreIterationList.add((Integer) sHist.get(rowInd, colIndices[0]));
+      _sc._scoringTimes.add(fmt.parseMillis((String) sHist.get(rowInd, colIndices[1])));
+      _sc._likelihoods.add((Double) sHist.get(rowInd, colIndices[2]));
+      _sc._objectives.add((Double) sHist.get(rowInd, colIndices[3]));
+      if (_parms._HGLM) {  // for HGLM family
+        _sc._convergence.add((Double) sHist.get(rowInd, colIndices[4]));
+        _sc._sumEtaiSquare.add((Double) sHist.get(rowInd, colIndices[5]));
+      }
+    }
+  }
+  
+  private int[] grabHeaderIndex(TwoDimTable sHist, int numHeaders, String[] colHeadersUseful) {
+    int[] colHeadersIndex = new int[numHeaders];
+    String[] colHeaders = sHist.getColHeaders();
+    for (int colInd = 0; colInd < numHeaders; colInd++) {
+      if (colInd == 0) {
+        int indexFound = Arrays.asList(colHeaders).indexOf(colHeadersUseful[colInd]);
```

Comment:
`Arrays.asList(colHeaders)` could be a local variable since it is used in both if branches

Reply:
Good point.